packages/jaeger-ui/src/components/SearchTracePage/index.js

```diff
@@ -62,7 +62,7 @@ export class SearchTracePageImpl extends Component {
       fetchMultipleTraces(needForDiffs);
     }
     fetchServices();
-    const { service } = store.get('lastSearch') || {};
+    const { service } = urlQueryParams || store.get('lastSearch') || {};
```

Comment:
I'm not a JS expert, but this looks odd to me - queryParams is an object, right? So wouldn't it evaluate to true and short circuit the rest even if the objects doesn't have a 'service' property?

I would like to see a unit tests.

Reply:
I think that queryParams can be undefined (if no query is passed) or an object otherwise. After this comment I played a bit with the UI and it seems that if a param is passed (i.e. `end`), then the `service` param is mandatory.

To be absolutely sure of this behavior though I updated the code to explicitly overwrite the `service` variable only if the `service` param is provided, in a way that supports both `undefined` and `{}` values of `urlQueryParams`.
I also added and updated unit test to verify that:
1. If `urlQueryParams={}`, the system fetches the operations for the stored service
2. If `urlQueryParams` includes `service`, the system works as expected

As I'm quite new to this project and to open source contributions in general, let me know if there's anything I should improve.

Thanks!